src/lib/provable/gadgets/arithmetic.ts

```diff
@@ -56,3 +56,55 @@ function divMod32(n: Field, nBits = 64) {
 function addMod32(x: Field, y: Field) {
   return divMod32(x.add(y), 33).remainder;
 }
+
+function divMod64(n: Field, nBits = 128) {
+  assert(
+    nBits >= 0 && nBits < 255,
+    `nBits must be in the range [0, 255), got ${nBits}`
```

Comment:
Is 255 not included because of Pasta curves?

Reply:
I am not sure why we don't include 255. I followed the range checking practice used in `divMod32`. Maybe @mitschabaude knows the reason?

Reply:
Because the field size is less than 2^255, there are 255 bit integers that have two different representations in the field (in fact that's true for almost all 255 bit integers since the field size is just slightly over 2^254)

That non-uniqueness would make the divmod gadget unsound: the prover could pick between two different quotient/remainder splits, one of which is wrong.